gaskij/rpicampusmap
# Button to login and backend architecture to logout

## Description
Allows CampusMap users to login with RPI CAS authentication using a login button on the front page. Users will be directed to the official CAS login page and then redirected back to CampusMap once a successful authentication is complete. Users can check their authentication status under the Profile tab. Logout capability is also part of this branch, but it requires going to the specific logout endpoint.

Fixes #(issue name)

## Solution
Created a Express session to keep track of user login information. Utilized the cas-authentication module for Node to communicate with RPI's CAS server. Set up a controller and route in the backend to handle login/logout/authenticate requests. The front end consists of adding a button to the header that communicates with the backend routes.

## Known Issues
None

## Testing Procedures
Try to login using the button and your CAS credentials. Check your login username is stored by clicking Profile. Then go to the api/cas/logout endpoint to see that you have been successfully logged out. Return to the CampusMap homepage and click on profile to double check that your session ended and you are no longer authenticated.

## Checklist for author
